packages/state-transition/src/cache/pubkeyCache.ts

```diff
@@ -1,11 +1,13 @@
 import {CoordType, PublicKey} from "@chainsafe/bls/types";
 import bls from "@chainsafe/bls";
+import * as immutable from "immutable";
```

Comment:
using the vanilla immutable implementation as it fits our needs at least API-wise. @wemeetagain @tuyennhv any consideration you want to mention on why we forked `persistent-ts` https://github.com/ChainSafe/ssz/tree/master/packages/persistent-ts ?